ToolBox/AlgoSeekOptionsConverter/AlgoSeekOptionsReader.cs

```diff
@@ -0,0 +1,225 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Logging;
+
+
+namespace QuantConnect.ToolBox.AlgoSeekOptionsConverter
+{
+	public class AlgoSeekOptionsReader : IDisposable
+	{
+		private const int LogInterval = 1000000;
+
+		private readonly string _file;
+		private readonly Stream _stream;
+		private readonly StreamReader _streamReader;
+		private readonly DateTime _referenceDate;
+
+		private BaseData _nextLine = null;
+
+		public bool HasNext { get { return _nextLine != null; } }
+
+		public BaseData Peek { get { return _nextLine; } }
+
+		public long Count { get; private set; }
+
+		private bool IsEOF { get { return _streamReader.Peek() == -1; } }
+
+		public AlgoSeekOptionsReader(string file)
+		{
+			_file = file;
+
+			var streamProvider = StreamProvider.ForExtension(Path.GetExtension(file));
+			_stream = streamProvider.Open(file).First();
+			_streamReader = new StreamReader(_stream);
+
+			_nextLine = GetNextValidLine();
```

Comment:
This should be MoveNext() which would set public property Current to follow IEnumerator pattern